Yosemite/Yosemite/PointOfSale/Items/POSProductOrVariationResolver.swift

```diff
@@ -13,61 +13,82 @@ struct POSProductOrVariationResolver {
         self.itemMapper = itemMapper ?? PointOfSaleItemMapper(currencySettings: currencySettings)
     }
 
-    func itemForProductOrVariation(_ productOrVariation: POSProduct) async throws -> POSItem {
+    func itemForProductOrVariation(_ productOrVariation: POSProduct,
+                                   scannedCode: String) async throws(PointOfSaleBarcodeScanError) -> POSItem {
```

Comment:
Good choice to use this error name here – with the persistence project, this whole file potentially goes away, but the errors are still likely to be valid.